snooty/parser.py

```diff
@@ -424,6 +438,60 @@ def dispatch_departure(self, node: docutils.nodes.Node) -> None:
                 target.children = [identifier]
                 item.term.append(target)
 
+    def handle_tabset(self, node: n.Directive, line: int) -> None:
+        tabset = node.options["tabset"]
+        # retrieve dictionary associated with this specific tabset
+        if tabset not in specparser.SPEC.tabs:
+            self.diagnostics.append(UnknownTabset(tabset, line))
+            print("unknown tabset!! ")
+            return
+
+        tab_definitions_list = specparser.SPEC.tabs[tabset]
```

Comment:
Idiomatic Python follows the [EAFP] (Easier to Ask Forgiveness than Permission) principle. That is, it's better to try to access this, then if it raises `KeyError`, raise a diagnostic.

[eafp]: https://docs.python.org/3/glossary.html#term-eafp